topics/dev/tutorials/tool-from-scratch/tutorial.md

```diff
@@ -0,0 +1,1182 @@
+1. Go to the "tools" directory in your Galaxy instance.
+2. Make a new directory for tools you will be wrapping.
+3. Use `planemo` to initialize a new tool wrapper with basic formatting.
+    - Here we will be using bellerophon, but the parameters can be set for any tool.
+    - The tool ID is the internal reference id for the tool, and may contain no spaces or capitalized letters
+    - The tool name is the text that will be visible to the user.
+
+    ```bash
+    cd galaxy/tools
+    mkdir -p myTools/bellerophon
+    cd myTools/bellerophon
+    planemo tool_init --id bellerophon --name bellerophon
+    ```
+
+4. You can now open the new xml file that has been generated to begin the wrapping. It should look something like this:
+![Bellerophon initial wrapper](../../images/tool_init.png "Initial tool wrapper after running planemo tool_init")
+
+## Galaxy Tool Wrappers
+
+Galaxy tool wrapper xml files are made up of several sections:
+
+- Tool
+- Description
+- Requirements
+- Macros
+- Command
+- Inputs
+- Outputs
+- Tests
+- Citations
+- Help
+
+While many of those are present in the wrapper by default, others need to be added.
+
+### Tool section
+
+The tool section at the head of the file defines several key pieces of information for both the user and Galaxy.
+The tool ID and name are defined here as well as which minimum version of Galaxy is required to run it.
+
+For this section, the following is necessary:
+
+- Tool ID
+- Tool name
+- Version (Version of the Galaxy wrapper)
+  - This is set as `'<Conda wrapper version>'+'galaxy<galaxy wrapper version>'`
+- Profile
+  - This is the minimum version of Galaxy required to run the tool
+- Unset python_template version
+
+when complete, the line should appear as follows:
+
+```xml
+<tool ID="bellerophon" name="bellerophon" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="20.01">
+```
+
+The @TOOL_VERSION@ and @VERSION_SUFFIX@ are what are referred to as "tokens", and will be furter discussed in the "Macros" section.
+
+### Description section
+
+This section adds help text to easily identify the tool. It is not intended as in-depth help, which should be in the `help` section.
+The description is simply presented as plaintext between the tags. Bellerophon's would look like this:
+
+```xml
+<description>chimeric reads from Arima Genomics</description>
+```
+
+Which would present the tool's in the toolbar and at the top of the tool form as
+
+```txt
+bellerophon: chimeric reads from Arima Genomics
+```
+
+### Macros section
+
+The macros section serves two functions -- to create meta-variables for inside both the xml text and the command text, called
+tokens, and xml sections that can be slotted in repeatedly. Although this tool will only use the tokens, the xml macros are very
+useful for tools which have multiple options/modes that reuse parameters.
+
+The @TOOL_VERSION@ and @VERSION_SUFFIX@ used in the Tool line are defined here. Each need their own 'token' line, resulting in
+
+```xml
```

Comment:
is this a hands on section too? or just documentation?

Reply:
Kind of both?

Reply:
Ok, then let's wrap it in a hands-on as well, just for consistency of noting which steps should be run.